Warn when running on unsupported platforms

[This CI run](https://github.com/LilithHafner/Chairmarks.jl/actions/runs/8239973789/job/22534386379) using Ubuntu & Julia 1.6 had these logs

```julia
   Testing Chairmarks
      Status `/tmp/jl_aHcgtG/Project.toml`
  [4c88cf16] Aqua v0.8.4
  [0ca39b1e] Chairmarks v1.1.2 `~/work/Chairmarks.jl/Chairmarks.jl`
  [2cca1867] RegressionTests v0.0.5
  [de0858da] Printf `@stdlib/Printf`
  [10745b16] Statistics `@stdlib/Statistics`
  [8dfed614] Test `@stdlib/Test`
      Status `/tmp/jl_aHcgtG/Manifest.toml`
  [4c88cf16] Aqua v0.8.4
  [0ca39b1e] Chairmarks v1.1.2 `~/work/Chairmarks.jl/Chairmarks.jl`
  [34da2185] Compat v4.14.0
  [2cca1867] RegressionTests v0.0.5
   [17] macro expansion
      @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [18] macro expansion
      @ ~/work/Chairmarks.jl/Chairmarks.jl/test/runtests.jl:351 [inlined]
   [19] macro expansion
      @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [20] top-level scope
      @ ~/work/Chairmarks.jl/Chairmarks.jl/test/runtests.jl:6
   [21] include(fname::String)
      @ Base.MainInclude ./client.jl:444
   [22] top-level scope
      @ none:6
   [23] eval
      @ ./boot.jl:360 [inlined]
   [24] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:261
   [25] _start()
      @ Base ./client.jl:485
Test Summary:          |   Pass  Error  Broken   Total
Chairmarks             | 754072      1       3  754076
  Standard tests       | 754046                 754046
  Statistics Extension |      2                      2
  Precision            |     14              2      16
  Performance          |                     1       1
  Aqua                 |     10                     10
  Regression Tests     |             1               1
ERROR: LoadError: Some tests did not pass: 754072 passed, 0 failed, 1 errored, 3 broken.
in expression starting at /home/runner/work/Chairmarks.jl/Chairmarks.jl/test/runtests.jl:5
ERROR: LoadError: Package Chairmarks errored during testing
Stacktrace:
 [1] pkgerror(msg::String)
   @ Pkg.Types /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Types.jl:55
 [2] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, julia_args::Cmd, test_args::Cmd, test_fn::Nothing)
   @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Operations.jl:1712
 [3] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, test_fn::Nothing, julia_args::Vector{String}, test_args::Cmd, kwargs::Base.Iterators.Pairs{Symbol, IOContext{Base.PipeEndpoint}, Tuple{Symbol}, NamedTuple{(:io,), Tuple{IOContext{Base.PipeEndpoint}}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:343
 [4] test(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOContext{Base.PipeEndpoint}, kwargs::Base.Iterators.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:coverage, :julia_args), Tuple{Bool, Vector{String}}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:80
 [5] test(; name::Nothing, uuid::Nothing, version::Nothing, url::Nothing, rev::Nothing, path::Nothing, mode::Pkg.Types.PackageMode, subdir::Nothing, kwargs::Base.Iterators.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:coverage, :julia_args), Tuple{Bool, Vector{String}}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:96
 [6] top-level scope
   @ ~/work/_actions/julia-actions/julia-runtest/v1/test_harness.jl:15
 [7] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [8] top-level scope
   @ none:1
in expression starting at /home/runner/work/_actions/julia-actions/julia-runtest/v1/test_harness.jl:7
Error: Process completed with exit code 1.
```

The problem was that it ran RegressionTests on an unsupported platform. The solution is to pass `skip_unsupported_platforms=true`. There should be a prominent warning and hint when this happens. It should be easy to implement this as we already check if the platform is supported.

--- a/src/RegressionTests.jl
+++ b/src/RegressionTests.jl
@@ -43,10 +43,15 @@ end
 is_platform_supported() = VERSION >= v"1.9" && !Sys.iswindows()
 
 function test(::Type{Bool}; skip_unsupported_platforms=false, kws...)
-    if skip_unsupported_platforms && !is_platform_supported()
-        @warn "Skipping regression tests on unsupported platform"
-        return true
+    if !is_platform_supported()
+        if skip_unsupported_platforms
+            @warn "Skipping regression tests on unsupported platform"
+            return true
+        else
+            @warn "Running regressiong tests on unsupported platform"
+        end
     end
+
     report_changes(runbenchmarks(project=dirname(pwd()), kws...))
 end
 struct RegressionTestFailure <: Exception end

